hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -183,9 +185,10 @@ public BlockOutputStream(
     // tell DataNode I will send incremental chunk list
     if (config.getIncrementalChunkList()) {
       this.containerBlockData.addMetadata(INCREMENTAL_CHUNK_LIST_KV);
-      this.lastChunkBuffer =
-          ByteBuffer.allocate(config.getStreamBufferSize());
+      this.lastChunkBuffer = DIRECT_BUFFER_POOL.getBuffer(config.getStreamBufferSize());
```

Comment:
Note: both `DirectBufferPool` methods appears to be thread-safe, that is good news and one less problem for HDDS-9844.